pyramid/config/views.py

```diff
@@ -933,7 +933,7 @@ def register(permission=permission, renderer=renderer):
             if not exception_only and isexc:
                 derived_view = runtime_exc_view(derived_view, derived_exc_view)
 
-            derived_view.__discriminator__ = lambda *arg: discriminator
+            derived_view.__discriminator__ = lambda *arg: view_intr.discriminator
```

Comment:
Yes, it was a first solution that I have tried when investigate this reference cycle. It works too.